app/code/Magento/Sales/etc/di.xml

```diff
@@ -1015,4 +1015,11 @@
     <preference
             for="Magento\Sales\Api\OrderCustomerDelegateInterface"
             type="Magento\Sales\Model\Order\OrderCustomerDelegate" />
+    <type name="Magento\Sales\Model\Order\Reorder\OrderedProductAvailabilityChecker">
+        <arguments>
+            <argument name="productAvailabilityChecks" xsi:type="array">
+                <item name="configurable" xsi:type="object">Magento\ConfigurableProductSales\Model\Order\Reorder\OrderedProductAvailabilityChecker</item>
+            </argument>
+        </arguments>
+    </type>
```

Comment:
Unfortunately, the current solution is not resolving the dependency issue between the modules.
I guess this definition still should be kept under **Magento_ConfigurableProductSales** (in `app/code/Magento/ConfigurableProductSales/etc/di.xml`).

And it should be enough to inject an empty array as default value of `productAvailabilityChecks` parameter (in `app/code/Magento/Sales/etc/di.xml`).
```
<type name="Magento\Sales\Model\Order\Reorder\OrderedProductAvailabilityChecker">
    <arguments>
        <argument name="productAvailabilityChecks" xsi:type="array" />
    </arguments>
</type>
```

Reply:
Fixed. Check pls.